tsz-compress/src/deltadelta_zigzag.rs

```diff
@@ -0,0 +1,589 @@
+use crate::prelude::{BitBuffer, BitBufferSlice};
+
+pub fn encode_deltadelta_i8(value: i8, out: &mut BitBuffer) {
+    out.push(false);
+    if value == 0 || value == -1 {
+        // 1 bit, header 00
+        out.push(false);
+        out.push(false);
+        if value == 0 {
+            out.push(false);
+        }
+        else {
+            out.push(true);
+        }
+        return;
+    } else if (-15..15).contains(&value){
+        // 5 bits, header 01
+        out.push(false);
+        out.push(true);
+        let encoded: u8 = ((value << 1) ^ (value >> 7)) as u8;
```

Comment:
```suggestion
        let encoded = ((value << 1) ^ (value >> 7)) as u8;
```

Prefer to leave the types inferred if they are specified on the right hand side of an assignment